File: compress-video.js
const Handbrake = require('handbrake-js');
const fs = require('fs');
const path = require('path');

const inputPath = path.join(__dirname, 'src/videos/Header-impression.mp4');
const outputPath = path.join(__dirname, 'src/videos/Header-impression-compressed.mp4');

// Get original file size
const originalSize = fs.statSync(inputPath).size / (1024 * 1024);
console.log(`Original size: ${originalSize.toFixed(2)} MB`);
console.log('Compressing video...\n');

// Compress the video
Handbrake.spawn({
  input: inputPath,
  output: outputPath,
  preset: 'Fast 1080p30',
  // Additional options for better compression
  'quality': 20,  // Quality (0-51, lower = better)
  'vb': 1500,     // Video bitrate (kbps)
  'ab': 96        // Audio bitrate (kbps)
})
.on('progress', (progress) => {
  process.stdout.write(`Progress: ${progress.percentComplete}%\r`);
})
.on('complete', () => {
  const compressedSize = fs.statSync(outputPath).size / (1024 * 1024);
  const reduction = ((originalSize - compressedSize) / originalSize) * 100;
  
  console.log('\n\n✓ Compression complete!');
  console.log(`Original size: ${originalSize.toFixed(2)} MB`);
  console.log(`Compressed size: ${compressedSize.toFixed(2)} MB`);
  console.log(`Reduction: ${reduction.toFixed(1)}%`);
  console.log(`Output: ${outputPath}`);
})
.on('error', (err) => {
  console.error('Error:', err);
  process.exit(1);
});

File: compress_video.py
import subprocess
import os

input_video = r"c:\Users\yuanw\github\GFL-Trunk\src\videos\Header-impression.mp4"
output_video = r"c:\Users\yuanw\github\GFL-Trunk\src\videos\Header-impression-compressed.mp4"

# Get original file size
original_size = os.path.getsize(input_video) / (1024 * 1024)
print(f"Original size: {original_size:.2f} MB")
print("Compressing video (this may take a few minutes)...")

# Use ffmpeg via subprocess - lower bitrate and codec optimization
cmd = [
    'ffmpeg',
    '-i', input_video,
    '-c:v', 'libx264',
    '-crf', '18',  # Quality (18-28, lower = better quality)
    '-preset', 'slow',  # Speed/quality tradeoff
    '-c:a', 'aac',
    '-b:a', '96k',  # Lower audio bitrate
    '-y',  # Overwrite output
    output_video
]

try:
    result = subprocess.run(cmd, capture_output=True, text=True)
    
    if result.returncode == 0:
        compressed_size = os.path.getsize(output_video) / (1024 * 1024)
        reduction = ((original_size - compressed_size) / original_size) * 100
        
        print(f"\n✓ Compression complete!")
        print(f"Original size: {original_size:.2f} MB")
        print(f"Compressed size: {compressed_size:.2f} MB")
        print(f"Reduction: {reduction:.1f}%")
        print(f"Output: {output_video}")
    else:
        print(f"Error: {result.stderr}")
except FileNotFoundError:
    print("ffmpeg not found. Please install FFmpeg and try again.")
